src/Facets/SquidFacet.sol

```diff
@@ -0,0 +1,133 @@
+    /// @param _squidData data specific to Squid Router
+    function startBridgeTokensViaSquid(
+        ILiFi.BridgeData memory _bridgeData,
+        SquidData memory _squidData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        validateBridgeData(_bridgeData)
+        doesNotContainSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+    {
+        LibAsset.depositAsset(
+            _bridgeData.sendingAssetId,
+            _bridgeData.minAmount
+        );
+
+        _startBridge(_bridgeData, _squidData);
+    }
+
+    /// @notice Swaps and bridges tokens via Squid Router
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _swapData an array of swap related data for performing swaps before bridging
+    /// @param _squidData data specific to Squid Router
+    function swapAndStartBridgeTokensViaSquid(
+        ILiFi.BridgeData memory _bridgeData,
+        LibSwap.SwapData[] calldata _swapData,
+        SquidData memory _squidData
+    )
+        external
+        payable
+        nonReentrant
+        refundExcessNative(payable(msg.sender))
+        containsSourceSwaps(_bridgeData)
+        doesNotContainDestinationCalls(_bridgeData)
+        validateBridgeData(_bridgeData)
+    {
+        _bridgeData.minAmount = _depositAndSwap(
+            _bridgeData.transactionId,
+            _bridgeData.minAmount,
+            _swapData,
+            payable(msg.sender)
+        );
+        _startBridge(_bridgeData, _squidData);
+    }
+
+    /// Internal Methods ///
+
+    /// @dev Contains the business logic for the bridge via Squid Router
+    /// @param _bridgeData the core information needed for bridging
+    /// @param _squidData data specific to Squid Router
+    function _startBridge(
+        ILiFi.BridgeData memory _bridgeData,
+        SquidData memory _squidData
+    ) internal {
+        IERC20 sendingAssetId = IERC20(_bridgeData.sendingAssetId);
+        uint256 msgValue = LibAsset.isNativeAsset(address(sendingAssetId))
+            ? _bridgeData.minAmount
+            : 0;
+
+        if (msgValue == 0) {
+            if (
+                _squidData.callType == SquidCallType.CallBridge ||
+                _squidData.callType == SquidCallType.CallBridgeCall
+            ) {
+                LibAsset.maxApproveERC20(
+                    sendingAssetId,
+                    address(squidMulticall),
+                    _bridgeData.minAmount
+                );
+            } else if (_squidData.callType == SquidCallType.BridgeCall) {
+                LibAsset.maxApproveERC20(
+                    sendingAssetId,
+                    address(squidRouter),
+                    _bridgeData.minAmount
+                );
+            }
+        }
+
+        (bool success, bytes memory res) = address(squidRouter).call{
+            value: msgValue
+        }(_squidData.callData);
```

Comment:
If the _bridgeData.minAmount for an ERC20 was adjusted we do not pass it here yet.